pkg/controller/master/virtualmachine/register.go

```diff
@@ -61,9 +62,10 @@ func Register(ctx context.Context, management *config.Management, _ config.Optio
 		vmBackupCache:  vmBackupCache,
 		snapshotClient: snapshotClient,
 		snapshotCache:  snapshotCache,
+		settingCache:   settingCache,
```

Comment:
it does not look we need the `settingCache` for anything other than initialise the `vmrCalculator`. we could drop this and reduce number of files being changed.

Reply:
Current code calls `kubevirtservices.GetMemoryOverhead(vmi, runtime.GOARCH, nil)` and uses nil as the ratio, from v1.4.0 the ratio configuration is introduced.

https://github.com/harvester/harvester/blob/3841bb7589abfc29f73912103e84af8ac88d35a1/pkg/util/resourcequota/calculator.go#L201

https://github.com/harvester/harvester/blob/3841bb7589abfc29f73912103e84af8ac88d35a1/pkg/util/resourcequota/calculator.go#L345

Now it becomes `memoryOverhead := kubevirtservices.GetMemoryOverhead(vmi, runtime.GOARCH, util.GetAdditionalGuestMemoryOverheadRatioWithoutError(c.settingCache))`

the settingCache is used